packages/loader/container-loader/src/test/connectionManager.spec.ts

```diff
@@ -216,12 +231,73 @@ describe("connectionManager", () => {
 			assert.strictEqual(connectionCount, 1, "Expect there to still not be a connection yet");
 			checkedTimeout = true;
 		}, 300);
-
 		connection = await waitForConnection();
 		assert.strictEqual(connectionCount, 2, "Expect there to be a connection after waiting");
 		assert(checkedTimeout, "Expected to have checked 300ms timeout");
 	});
 
+	it("Does not re-try connection on error if ReconnectMode=Disabled", async () => {
+		const connectionManager = createConnectionManager();
+		// mock connectToDeltaStream method so that it throws a retriable error when connect() is called in connectionManager
+		const stubbedConnectToDeltaStream = stub(mockDocumentService, "connectToDeltaStream");
+		const retryAfter = 0.2; // seconds
+		stubbedConnectToDeltaStream.throws(
+			// Throw retryable error
+			new RetryableError("Test message", NackErrorType.ThrottlingError, {
```

Comment:
it would be great to add some counter and count every call, such that UT code can validate that it actually spins, and then it stops spinning once disconnect is called.

Reply:
The the current two tests test whether re-connect is attempted for both reconnectMode enabled and disabled - which should suffice testing for testing of connection manager behavior for now. (to set the reconnect mode, we  the `setAutoReconect` which is the method invoked by the container internally.)

I agree it would be more accurate to have a test at the "container" level, where we directly called connect() and disconnect() on the container and test reconnect behavior, but I couldn't find an easy way of testing with real `Container` object without mocking almost every parameter that the container requires. - Open to ideas/code pointers here.